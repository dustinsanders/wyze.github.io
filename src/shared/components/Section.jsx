// @flow

import type { Style, VNode } from '../../types'
import { css } from 'glamor'
import preact from 'preact'

type Props = {
  center?: boolean,
  children?: VNode,
  styles?: Style,
}

const centered: Style = css({
  textAlign: 'center',
})

const Section = ({ center, children, styles }: Props): VNode => (
  <div {...css(center && centered, styles)}>
    {children}
  </div>
)

Section.defaultProps = {
  center: true,
  children: null,
  styles: {},
}

export default Section
