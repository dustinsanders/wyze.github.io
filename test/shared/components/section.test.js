// @flow

import Section from '../../../src/shared/components/Section'
import preact from 'preact'
import snapshot from '../../helpers/snapshot'

const children = (
  // eslint-disable-next-line react/jsx-filename-extension
  <div>I am a child.</div>
)

const styles = {
  color: 'rebeccapurple',
}

describe('<Section />', () => {
  it('renders', snapshot.bind(null, Section, { children }))
  it('renders non-centered', snapshot.bind(null, Section, { center: false }))
  it('renders passed styles', snapshot.bind(null, Section, { styles }))
})
