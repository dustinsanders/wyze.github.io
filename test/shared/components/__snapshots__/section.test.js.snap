// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> renders 1`] = `
"<div data-css-1yv2jt6>
  <div>I am a child.</div>
</div>"
`;

exports[`<Section /> renders non-centered 1`] = `"<div></div>"`;

exports[`<Section /> renders passed styles 1`] = `"<div data-css-1op8xq1></div>"`;
